build/hostlogswindows/installer/scripts/livenessprobe.ps1

```diff
@@ -0,0 +1,30 @@
+param(
+    [Parameter( Position=0,
+                mandatory=$true,
+                HelpMessage="Path of the exeutable you want to check if it is running. Be sure to provide '\\' when entering the path.")] 
+    [string]$exeRelativePath,
+
+    [Parameter( Position=1,
+                mandatory=$true,
+                HelpMessage="Relative path of the file you want to check if it exists.")] 
+    [string]$fileName
+)
+
+$SUCCESS=0
+$FILESYSTEM_WATCHER_FILE_EXISTS=1
+$NO_MONAGENT_MANAGER_PROCESS=2
+
+if (Test-Path -Path $fileName -PathType Leaf) 
+{
+    Write-Error "INFO: File: $filename exists indicates Config Map Updated since agent started."
+    exit $FILESYSTEM_WATCHER_FILE_EXISTS
+}
+
+if($Env:HOSTLOGS_MA_STARTED -eq "true")

```

Comment:
Without it the livenessprobe thinks the container crashed because the exe doesn't execute unless the environment variables already exist. So instead, we first check to see if we should even expect the exe to up yet. 

If main.ps1 fails in anyway the container will crash since it is the entrypoint for the container image.

Reply:
> How about tweaking `initialDelaySeconds` in the liveness probe setup to give the MA enough start time? Relying on a global env var at the host level might get tricky, particularly if it sticks around after the container's life ends.

How long is long enough for initialDelaySeconds though? On a node which is extremely low on resources the MA could take a while to start up. If the delay is not long enough we could end up repeatedly crashing a healthy pod. But we also don't want to wait too long to restart an unhealthy pod

Reply:
Once we have containerd 1.7 a user level variable would work well here. The plan is to create a new admin user for each pod, so a user level variable would be available to all processes in the pod, but should be cleaned up when the pod dies

Reply:
Are slow starts something we have observed or is this speculation? I would expect the agent to become present in the process list fairly quickly even if it is taking a while to startup completely, no? 

In any case, we should have an understanding of a worst-case startup time and have a timeout somewhere. The current implementation doesnt protect against long startup times anyway since the monitored process is started in the background and the env variable is set immediately (even potentially before the process actually starts). 

If we want to guard against an issue with slow starts and the process not becoming viewable in the process list before the livenessProbe kicks in, consider using a startupProbe with the same check: https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-startup-probes

Reply:
I haven't encountered slow starts with this setup and the initalDelaySeconds at 60 should help with that. If MonAgentLauncher.exe doesnt start up by that point then we should reset the container. 

The startupprobe could check to see if the executable is up and running but we run into the same issue if the configmap isn't added to the cluster yet then it isn't going to be return healthy. Unlike the livenessprobe the startupprobe only runs once so the customer only get once chance to get it on the cluster before it restart the container.

Reply:
Instead of using a flag I was able to use placeholder values for the needed env vars and execute the monagent with those. I tested to see

1) Does the container restart when I stop the process - yes

2) Does the container use the new env vars when the config map is changed - yes

3) Does the env vars go away on reset - yes

Reply:
I think there's 2 different problems we're trying to solve here.

1. If AMA takes a while to start up it could (theoretically) trigger a liveness probe crash – I think the startup probe is probably the right solution to this
2. Previously we were not starting the monitoring agent if it had not been configured via ConfigMap. The env var was being used to determine this, otherwise we'd always get an infinite crash loop until the ConfigMap was provided – Adding placeholder values should address this

So maybe we should do both placeholder values + startup probe